src/main/java/com/example/hop_oasis/service/data/UserAuthenticated.java

```diff
@@ -0,0 +1,25 @@
+package com.example.hop_oasis.service.data;
+
+import com.example.hop_oasis.model.User;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserAuthenticated {
```

Comment:
This is not a proper name for a class. Class name should be a noun, and it's more like a question)
Consider `AuthenticationContext`, or smth else)
I also like `IdentityProvider`